### PR TITLE
Add support for kernel crypto API in user space

### PR DESCRIPTION
**Pull request by niontive**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

1) Add kernel configs to support crypto API in user space
2) Add libkcapi package, which is a user-space lib for the kernel crypto API
3) Use libkcapi to HMAC calc the kernel (required for future FIPS work)

###### Change Log  

- Add kernel configs to support crypto API in user space
- Add libkcapi package
- Use libkcapi to HMAC calc the kernel (required for future FIPS work)

###### Does this affect the toolchain?  

Yes

###### Associated issues  

None

###### Links to CVEs  
None

###### Test Methodology

Local build

### CODE REVIEW

File: SPECS/libkcapi/libkcapi.spec [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+%{fipshmac} -d "$lib_path"/fipscheck                               \\\
+  "$lib_path"/libkcapi.so.%{version} || exit 1                   \
+ln -s libkcapi.so.%{version}.hmac                            \\\
+  "$lib_path"/fipscheck/libkcapi.so.%{vmajor}.hmac               \
+%{nil}
+# Do we build the replacements packages?
+%bcond_with replace_coreutils
+%bcond_without replace_fipscheck
+%bcond_without replace_hmaccalc
+%bcond_without test_package
```
Comment (PawelWMS) [inline SPECS/libkcapi/libkcapi.spec:57]: We're not running our builds using the `--with-*` or `--without-*` switches. I'd suggest removing all `%bcond_*` macros and cleaning-up the spec accordingly to limit the noise.

File: SPECS/libkcapi/libkcapi.spec [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+pushd test
+ENABLE_FUZZ_TEST=1 \
+NO_32BIT_TEST=1    \
+  ./test-invocation.sh
+popd
+%endif
+%endif
+
+
+%ldconfig_scriptlets
```
Comment (PawelWMS) [inline SPECS/libkcapi/libkcapi.spec:295]: As far as I can remember, the `%ldconfig*` macros don't work on `1.0-dev` and have to be replaced with this pair:
```
%post -p /sbin/ldconfig
%postun -p /sbin/ldconfig
```

File: SPECS/libkcapi/libkcapi.spec [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+
+%if %{with test_package}
+%files tests
+%{_libexecdir}/%{name}/*
+%endif
+
+
+%changelog
+* Tue Jan 19 2021 Nicolas Ontiveros <niontive@microsoft.com> - 1.2.0-4
+- Initiali CBL-Mariner import from Fedora 33.
```
Comment (PawelWMS) [inline SPECS/libkcapi/libkcapi.spec:354]: Please verify the license and reflect that in the changelog (we usually use the "License verified" phrase).

Comment (PawelWMS) [inline SPECS/libkcapi/libkcapi.spec:354]: Please indicate the license for the Fedora spec (it's MIT). There's also a typo in "Initial".;) This line should say `Initial CBL-Mariner import from Fedora 33 (license: MIT).` in short.

**Review by danmihai1 — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**